pypeit/bspline/bspline.py

```diff
@@ -145,124 +131,168 @@ def __init__(self, x, fullbkpt=None, nord=4, npoly=1, bkpt=None, bkspread=1.0, v
             self.xmax= None
             self.funcname= None
             return
+
```

Comment:
Pretty significant changes here.  Do we have test coverage on this part of the code?  If not, can we (and run it on both the original and new)?